### PR TITLE
FEATURE: phase 1 of supporting multiple email addresses

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

**Comment by tgxworld** [comment]
Main ruby tests are passing but our plugin tests are broken.

File: app/models/invite.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ class UserExists < StandardError; end
   def user_doesnt_already_exist
     @email_already_exists = false
     return if email.blank?
-    u = User.find_by("email = ?", Email.downcase(email))
+    u = User.find_by_email(Email.downcase(email))
```
Comment (jjaffeux) [inline app/models/invite.rb]: Why are we doing `Email.downcase` two times ? (multiple instances of this in the commit)

https://github.com/discourse/discourse/blob/master/app/models/user.rb#L235

File: app/models/invite.rb [hunk 2]
```diff
@@ -98,11 +98,9 @@ def self.create_invite_by_email(email, invited_by, opts=nil)
     group_ids = opts[:group_ids]
     send_email = opts[:send_email].nil? ? true : opts[:send_email]
     custom_message = opts[:custom_message]
-
     lower_email = Email.downcase(email)
```
Comment (jjaffeux) [inline app/models/invite.rb]: Shouldn’t be needed? (see other comments)

File: app/models/invite.rb (hunk 1 above)
Comment (tgxworld) [inline app/models/invite.rb]: good catch! :+1:

File: app/models/email_token.rb [hunk 3]
```diff
@@ -71,7 +71,7 @@ def self.confirm(token)
       end
 
       if user
-        return User.find_by(email: Email.downcase(user.email)) if Invite.redeem_from_email(user.email).present?
+        return User.find_by_email(Email.downcase(user.email)) if Invite.redeem_from_email(user.email).present?
```
Comment (jjaffeux) [inline app/models/email_token.rb]: `Email.downcase` probably not needed here

**Pull request merged**